engine/macros.h

```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+#define to_hex(x)                                                              \
```

Comment:
why .h file,  I think use .hpp in engine directory.

Reply:
done